target/linux/siflower/files-6.12/drivers/gpio/gpio-siflower.c

```diff
@@ -317,12 +317,11 @@ static int sf_gpio_probe(struct platform_device *pdev)
 	return devm_gpiochip_add_data(dev, gc, priv);
 }
 
-static int sf_gpio_remove(struct platform_device *pdev)
+static void sf_gpio_remove(struct platform_device *pdev)
```

Comment:
move this to remove_new first in a separate commit so we can reduce patch delta and also align 6.6 to this change. hope it's not a problem for you!